proteingym/baselines/esm/compute_fitness.py

```diff
@@ -53,34 +39,77 @@ def read_msa(filename: str, nseq: int, sampling_strategy: str, random_seed: int,
         nseq = min(len(msa),nseq)
         msa = random.sample(msa, nseq)
     elif sampling_strategy=='sequence-reweighting':
-        MSA = MSA_processing(
-            MSA_location=filename,
-            use_weights=True,
-            weights_location=weight_filename
-        )
-        print("Neff: "+str(MSA.Neff))
-        print("Name of focus_seq: "+str(MSA.focus_seq_name))
-        all_sequences_msa=[]
-        weights=[]
-        msa=[]
-        for seq_name in MSA.raw_seq_name_to_sequence.keys():
-            if seq_name == MSA.focus_seq_name:
-                msa.append((seq_name,MSA.raw_seq_name_to_sequence[seq_name]))
-                del MSA.seq_name_to_weight[seq_name]
-            else:
-                if seq_name in MSA.seq_name_to_weight:
-                    all_sequences_msa.append((seq_name,MSA.raw_seq_name_to_sequence[seq_name]))
-                    weights.append(MSA.seq_name_to_weight[seq_name])
-        if len(all_sequences_msa)>0:
-            weights = np.array(weights) / np.array(list(MSA.seq_name_to_weight.values())).sum()
-            print("Check sum weights MSA: "+str(weights.sum()))
-            msa.extend(random.choices(all_sequences_msa, weights=weights, k=nseq-1))
+        # If MSA has already been processed, just use it here
+        if processed_msa is None:
+            if weight_filename is None:
+                print("Need weight filename if using sequence-reweighting sample strategy")
+            MSA = MSA_processing(
+                MSA_location=filename,
+                use_weights=True,
+                weights_location=weight_filename
+            )
+            print("Name of focus_seq: "+str(MSA.focus_seq_name))
+        else:
+            MSA = processed_msa
+
+        # Make sure we always keep the WT in the subsampled MSA
+        msa = [(MSA.focus_seq_name,MSA.raw_seq_name_to_sequence[MSA.focus_seq_name])]
+
+        non_wt_weights = np.array([w for k, w in MSA.seq_name_to_weight.items() if k != MSA.focus_seq_name])
+        non_wt_sequences = [(k, s) for k, s in MSA.seq_name_to_sequence.items() if k != MSA.focus_seq_name]
+        non_wt_weights = non_wt_weights / non_wt_weights.sum() # Renormalize weights
+
+        # Sample the rest of the MSA according to their weights
+        if len(non_wt_sequences) > 0:
+            msa.extend(random.choices(non_wt_sequences, weights=non_wt_weights, k=nseq-1))
+
+        # This is intended to filter out weights, but after hhfiltering our msa is already the correct set
```

Comment:
This comment refers to the old line 66, where we were filtering out weights but with a small possible bug - if we recalculate weights based on the hhfilter set then we won't need to filter weights out post-calc from the bigger MSA